### PR TITLE
Complete auth, return and tests

### PR DESCRIPTION
**Pull request by Trimnosh**

### CODE REVIEW

File: examples/RatepayInstall/return.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        echo 'Reservation';
+    } else {
+        echo 'Payment';
+    }
+    echo ' successfully completed.<br>';
+    echo getTransactionLink($baseUrl, $ratepayInstallMAID, $response->getTransactionId());
+    ?>
+    <br>
+    <?php
+    if ($response->getTransactionType() !== 'authorization-only') {
```
Comment (horfic) [inline examples/RatepayInstall/return.php]: no auth only supported for ratepay installment

File: src/Transaction/RatepayInstallTransaction.php [hunk 2]
```diff
@@ -91,12 +93,17 @@ public function setOrderNumber($orderNumber)
...
     protected function mappedSpecificProperties()
     {
-        return [
+        $result = [
             'order-number' => $this->orderNumber,
             'order-items' => $this->itemCollection->mappedProperties(),
             'cancel-redirect-url' => $this->redirect->getCancelUrl(),
             'success-redirect-url' => $this->redirect->getSuccessUrl(),
         ];
+        if ($this->redirect->getFailureUrl()) {
```
Comment (horfic) [inline src/Transaction/RatepayInstallTransaction.php]: why is failureurl seperat, not needed if cancel and success url is needed?

File: src/TransactionService.php [hunk 3]
```diff
@@ -145,6 +145,10 @@ public function handleResponse(array $payload)
             return $this->processAuthFrom3DResponse($payload);
         }
 
+        if (array_key_exists('base64payload', $payload) && array_key_exists('psp_name', $payload)) {
+            return $this->processRatepayInstallmentResponse($payload);
```
Comment (horfic) [inline src/TransactionService.php]: responsemapper map handles base64 content already, no extra function needed

**Comment by horfic** [comment]
crm id is also missing in example

File: src/TransactionService.php (hunk 3 above)
Comment (Trimnosh) [inline src/TransactionService.php]: Decoding removed.

**Pull request merged**